### PR TITLE
[emojicoin] Add fees

### PR DESCRIPTION
**Pull request by gregnazario**

### CODE REVIEW

**Comment by llamatester** [comment]
The emojicoin.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees emojicoin.ts
    
    🦙 Running EMOJICOIN.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 26 Nov 2024 00:00:00 GMT
    End Date:	Wed, 27 Nov 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 3/4/2023
    NO METHODOLOGY SPECIFIED
    Total fees: 2547.51 B
    Daily fees: 8.00 B
    End timestamp: 1732665599 (2024-11-26T23:59:59.000Z)

**Comment by llamatester** [comment]
The emojicoin.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees emojicoin.ts
    
    🦙 Running EMOJICOIN.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Tue, 26 Nov 2024 00:00:00 GMT
    End Date:	Wed, 27 Nov 2024 00:00:00 GMT
    ---------------------------------------------------
    
    APTOS 👇
    Backfill start time: 3/4/2023
    NO METHODOLOGY SPECIFIED
    Total fees: 304.94 k
    Daily fees: 958
    End timestamp: 1732665599 (2024-11-26T23:59:59.000Z)

File: helpers/aptops.ts [hunk 1]
```diff
@@ -1,22 +1,29 @@
-import { httpGet } from "../utils/fetchURL";
+import {httpGet, httpPost} from "../utils/fetchURL";
 
 export const APTOS_PRC = 'https://aptos-mainnet.pontem.network';
```
Comment (xbtmatt) [inline helpers/aptops.ts:3]:
```suggestion
export const APTOS_RPC = 'https://aptos-mainnet.pontem.network';
```

File: helpers/aptops.ts [hunk 2]
```diff
@@ -1,22 +1,29 @@
...
-    lastData = res.data
-    data.push(...lastData)
-    cursor = res.headers['x-aptos-cursor']
-  } while (lastData.length === 9999)
-  return data
+    const data: any = []
+    let lastData: any;
+    let cursor
+    do {
+        let url = `${APTOS_PRC}/v1/accounts/${account}/resources?limit=9999`
```
Comment (xbtmatt) [inline helpers/aptops.ts:10]:
```suggestion
        let url = `${APTOS_RPC}/v1/accounts/${account}/resources?limit=9999`
```

File: fees/emojicoin.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        return parseInt((BigInt(fees) / BigInt(10 ** 8)).toString(10)) * apt_price
+    }
+
+    const totalFees = parseFees(BigInt(todayRegistry.cumulative_integrator_fees.value));
+    const dailyFees = parseFees(BigInt(todayRegistry.cumulative_integrator_fees.value) - BigInt(yesterdayRegistry.cumulative_integrator_fees.value));
+
+    return {
+        totalFees: `${totalFees}`,
+        dailyFees: `${dailyFees}`,
+        // TODO: revenue
```
Comment (xbtmatt) [inline fees/emojicoin.ts:113]: The revenue can just be the integrator fees (what you're already using for total/daily)

Comment (0xngmi) [reply fees/emojicoin.ts:113]: so fees = revenue?

File: fees/emojicoin.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        "0xface729284ae5729100b3a9ad7f7cc025ea09739cd6e7252aff0beb53619cafe::emojicoin_dot_fun::registry_view",
+        [],
+        [],
+        version);
+    return result
+}
+
+const ONE_DAY: number = (24 * 60 * 60 * 1000);
+
+const fetch = async (timestamp: number) => {
```
Comment (g1nt0ki) [inline fees/emojicoin.ts:90]: change this to adapter version 2, then you get start and endTimestamp as part of input

File: fees/emojicoin.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    // Find the timestamp
+    const date = new Date(timestamp * 1000)
+    const closestToDate = await timestampToVersion(timestamp)
+    const previousDayTimestamp = new Date(date.getTime() - ONE_DAY).getTime() / 1000
+    const closestToPreviousDate = await timestampToVersion(previousDayTimestamp)
+
+    const yesterdayRegistry = await registryView(BigInt(closestToPreviousDate))
+    const todayRegistry = await registryView(BigInt(closestToDate))
+
+    const prices = await getPrices(["coingecko:aptos"], timestamp)
```
Comment (g1nt0ki) [inline fees/emojicoin.ts:100]: avoid pulling token prices and change code to something like [this](https://github.com/DefiLlama/dimension-adapters/blob/master/fees/milkyway.ts#L23)

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
.

**Pull request closed without merging**